pkg/endpoint/policy.go

```diff
@@ -529,6 +529,12 @@ func (e *Endpoint) regeneratePolicy(owner Owner, opts models.ConfigurationMap) (
 		labelsMap = e.prevIdentityCache
 	}
 
+	// Containers without a security identity are not accessible
+	if e.SecurityIdentity == nil {
+		e.getLogger().Warn("Endpoint lacks identity, skipping policy calculation")
+		return false, nil
+	}
+
```

Comment:
Can you confirm moving this code here is safe to do with the logic on lines 545-554? I.e., is the logic: 
```
!e.forcePolicyCompute && e.nextPolicyRevision >= revision && labelsMap == e.prevIdentityCache && opts == nil
```

always guaranteed to evaluate to `false` if `e.SecurityIdentity == nil`?

Reply:
Does it matter? If `e.SecurityIdentity == nil`, we'll generate a `DROP_ALL` BPF program, so it doesn't seem like any of those parameters (revision, ID cache, config) matter?

Reply:
Makes sense.